presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -102,46 +96,82 @@ public DiscoveryNodeManager(
             InternalCommunicationConfig internalCommunicationConfig)
     {
         this.serviceSelector = requireNonNull(serviceSelector, "serviceSelector is null");
-        this.nodeInfo = requireNonNull(nodeInfo, "nodeInfo is null");
         this.failureDetector = requireNonNull(failureDetector, "failureDetector is null");
         this.expectedNodeVersion = requireNonNull(expectedNodeVersion, "expectedNodeVersion is null");
         this.httpClient = requireNonNull(httpClient, "httpClient is null");
         this.nodeStateUpdateExecutor = newSingleThreadScheduledExecutor(threadsNamed("node-state-poller-%s"));
         this.httpsRequired = internalCommunicationConfig.isHttpsRequired();
-        this.currentNode = refreshNodesInternal();
+
+        PrestoNode currentNode = null;
+        for (ServiceDescriptor service : serviceSelector.selectAllServices()) {
+            URI uri = getHttpUri(service);
+            NodeVersion nodeVersion = getNodeVersion(service);
+            if (uri != null && nodeVersion != null) {
+                PrestoNode node = new PrestoNode(service.getNodeId(), uri, nodeVersion, isCoordinator(service));
+
+                if (node.getNodeIdentifier().equals(nodeInfo.getNodeId())) {
```

Comment:
This line is incredibly confusing. The previous location had this comment '// record current node' that was useful.